runtime/integration-tests/src/cases/liquidity_pools_gateway_queue.rs

```diff
@@ -1,41 +1,128 @@
+use cfg_primitives::currency_decimals;
 use cfg_traits::liquidity_pools::MessageQueue as MessageQueueT;
 use frame_support::assert_ok;
+use pallet_liquidity_pools::Message;
+use pallet_liquidity_pools_gateway::message::GatewayMessage;
+use runtime_common::account_conversion::AccountConverter;
 use sp_runtime::traits::One;
 
 use crate::{
+	cases::liquidity_pools::{
+		utils::*, AUSD_CURRENCY_ID, DEFAULT_BALANCE_GLMR, DEFAULT_DOMAIN_ADDRESS_MOONBEAM,
+		DOMAIN_MOONBEAM, GLMR_CURRENCY_ID, POOL_ID,
+	},
 	config::Runtime,
 	env::{Blocks, Env},
 	envs::fudge_env::{FudgeEnv, FudgeSupport},
-	utils::{currency::cfg, genesis, genesis::Genesis},
+	utils::{accounts::Keyring, currency::cfg, genesis, genesis::Genesis},
 };
 
-#[test_runtimes(all)]
-fn submit_and_process<T: Runtime + FudgeSupport>() {
-	let mut env = FudgeEnv::<T>::from_parachain_storage(
-		Genesis::default()
-			.add(genesis::balances::<T>(cfg(1_000)))
-			.storage(),
-	);
-
-	let expected_event = env.parachain_state_mut(|| {
-		let message = pallet_liquidity_pools::Message::AddPool { pool_id: 1 };
-		let nonce = <T as pallet_liquidity_pools_gateway_queue::Config>::MessageNonce::one();
-
-		assert_ok!(
-			<pallet_liquidity_pools_gateway_queue::Pallet<T> as MessageQueueT>::submit(
-				message.clone()
-			)
+mod inbound {
+	use super::*;
+
+	/// This test is basically `increase_deposit_request` but instead of
+	/// handling the message directly via the LP pallet, we submit it via
+	/// the LP Gateway Queue and confirm that it's processed accordingly.
+	#[test_runtimes([development])]
+	fn success<T: Runtime + FudgeSupport>() {
+		let mut env = FudgeEnv::<T>::from_parachain_storage(
```

Comment:
I think we need here fudge to generate a correct `MessageExecutionSuccess` by the router, right?

Reply:
I'll give it a try without.

Reply:
We have 2 tests now for LP gateway queue - 1 for inbound, 1 for outbound, and both confirm that the message is reaching the correct destination - LP for inbound, LP Gateway for outbound.